crates/sui-indexer-alt/migrations/2024-12-03-011506_coin_balance_buckets/up.sql

```diff
@@ -0,0 +1,28 @@
+-- A table of coin balance buckets, keyed on object ID and checkpoint sequence number.
+-- At the end of each checkpoint, we insert a row for each coin balance bucket, if it has changed.
+-- We bucketize coin balances to reduce the number of distinct values and help with both write and
+-- read performance. Bucket is calculated as floor(log10(coin_balance)).
+-- We also keep a record when we delete or wrap an object, which we would need for consistency query.
+-- All fields except the primary key will be `NULL` for delete/wrap records.
+CREATE TABLE IF NOT EXISTS coin_balance_buckets
+(
+    object_id                   BYTEA         NOT NULL,
+    cp_sequence_number          BIGINT        NOT NULL,
+    -- The kind of owner of this coin. We need this to support ConsensusV2 objects.
+    -- A coin can be either owned by an address through fast-path ownership, or
+    -- by an anddress through ConsensusV2 ownership.
+    -- Its definition is in `StoredOwnerKind` in `models/objects.rs`.
+    owner_kind                  SMALLINT,
```

Comment:
Some small changes based on @aschran's PR

```suggestion
    -- by an address through ConsensusV2 ownership.
    -- Its definition is in `StoredCoinOwnerKind` in `models/objects.rs`.
    coin_owner_kind             SMALLINT,
```